# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: 721d513e7080
# Full Hash: 721d513e7080984eb470427c174102d860b9c56a
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-02-23 09:38:00
# Description:
#   Bug 1755698 - Ensure we destroy HTMLCanvasElement's members during cycle collection unlinking. r=gfx-reviewers,lsalzman
#   
#   OffscreenCanvasDisplayHelper may call back into HTMLCanvasElement after
#   it has been unlinked and violate our assumption that
#   HTMLCanvasElement::InvalidateCanvasPlaceholder is only called when we
# ==============================================================================

diff -r 5a2b2968b2a9 -r 721d513e7080 dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Tue Feb 22 23:56:28 2022 +0000
+++ b/dom/html/HTMLCanvasElement.h	Wed Feb 23 00:20:01 2022 +0000
@@ -309,6 +309,7 @@
 
  protected:
   virtual ~HTMLCanvasElement();
+  void Destroy();
 
   virtual JSObject* WrapNode(JSContext* aCx,
                              JS::Handle<JSObject*> aGivenProto) override;
